library/spdm_responder_lib/libspdm_rsp_key_pair_info.c

```diff
@@ -0,0 +1,152 @@
+                                               response_size, response);
+    }
+
+    if (spdm_context->response_state != LIBSPDM_RESPONSE_STATE_NORMAL) {
+        return libspdm_responder_handle_response_state(spdm_context,
+                                                       spdm_request->header.request_response_code,
+                                                       response_size, response);
+    }
+
+    if (request_size < sizeof(spdm_get_key_pair_info_request_t)) {
+        return libspdm_generate_error_response(spdm_context,
+                                               SPDM_ERROR_CODE_INVALID_REQUEST, 0,
+                                               response_size, response);
+    }
+
+    key_pair_id = spdm_request->key_pair_id;
+    if (key_pair_id == 0) {
+        return libspdm_generate_error_response(spdm_context,
+                                               SPDM_ERROR_CODE_INVALID_REQUEST, 0,
+                                               response_size, response);
+    }
+
+    if (spdm_context->connection_info.connection_state <
+        LIBSPDM_CONNECTION_STATE_NEGOTIATED) {
+        return libspdm_generate_error_response(
+            spdm_context,
+            SPDM_ERROR_CODE_UNEXPECTED_REQUEST, 0,
+            response_size, response);
+    }
+
+    if (spdm_context->last_spdm_request_session_id_valid) {
+        session_info = libspdm_get_session_info_via_session_id(
+            spdm_context,
+            spdm_context->last_spdm_request_session_id);
+        if (session_info == NULL) {
+            return libspdm_generate_error_response(
+                spdm_context,
+                SPDM_ERROR_CODE_UNEXPECTED_REQUEST, 0,
+                response_size, response);
+        }
+        session_state = libspdm_secured_message_get_session_state(
+            session_info->secured_message_context);
+        if (session_state != LIBSPDM_SESSION_STATE_ESTABLISHED) {
+            return libspdm_generate_error_response(
+                spdm_context,
+                SPDM_ERROR_CODE_UNEXPECTED_REQUEST, 0,
+                response_size, response);
+        }
+    }
+
+    if (!libspdm_is_capabilities_flag_supported(
+            spdm_context, false, 0,
+            SPDM_GET_CAPABILITIES_RESPONSE_FLAGS_GET_KEY_PAIR_INFO_CAP)) {
+        return libspdm_generate_error_response(
+            spdm_context, SPDM_ERROR_CODE_UNSUPPORTED_REQUEST,
+            SPDM_GET_KEY_PAIR_INFO, response_size, response);
+    }
+
+    LIBSPDM_ASSERT(*response_size >= sizeof(spdm_key_pair_info_response_t));
+    public_key_info_len = (uint16_t)(*response_size - sizeof(spdm_key_pair_info_response_t));
+    libspdm_zero_mem(response, *response_size);
+
+    uint8_t *public_key_info;
+    public_key_info = (uint8_t*)response + sizeof(spdm_key_pair_info_response_t);
+    result = libspdm_read_key_pair_info(
+        spdm_context,
+        key_pair_id,
+        spdm_context->connection_info.algorithm.base_asym_algo,
```

Comment:
Can probably remove this parameter. Key pairs are independent of the currently negotiated algorithm.

Reply:
Hi all. I have some confused. @jyao1  @steven-bellock  In the SPDM spec, 
```
This request and its response shall report information for all key pairs on the Responder
independent of any negotiated parameters of the current SPDM connection.
```

But in the `Table 103 — KEY_PAIR_INFO response message format`,
![image](https://github.com/user-attachments/assets/60013914-309a-4d1e-8321-882ba38437fc)

Do we need to check if the CurrentAsymAlgo is configured correctly?

https://github.com/DMTF/libspdm/pull/2771#discussion_r1690685506

Reply:
>Do we need to check if the CurrentAsymAlgo is configured correctly?

Not right now. That check would go in `libspdm_check_context` and it would make sure that the algorithms in `CurrentAsymAlgo` match the ones in the Responder's local context for `BaseAsymSel`.